client/src/main/scala/org/http4s/client/middleware/Metrics.scala

```diff
@@ -44,35 +44,38 @@ object Metrics {
       ops: MetricsOps[F],
       classifierF: Request[F] => Option[String])(
       req: Request[F])(implicit F: Sync[F], clock: Clock[F]): Resource[F, Response[F]] =
-    (for {
+    for {
       statusRef <- Resource.liftF(Ref.of[F, Option[Status]](None))
       start <- Resource.liftF(clock.monotonic(TimeUnit.NANOSECONDS))
-      _ <- Resource.make(ops.increaseActiveRequests(classifierF(req)))(_ =>
-        ops.decreaseActiveRequests(classifierF(req)))
-      _ <- Resource.make(F.unit) { _ =>
-        clock
-          .monotonic(TimeUnit.NANOSECONDS)
-          .flatMap(now =>
-            statusRef.get.flatMap(oStatus =>
-              oStatus.traverse_(status =>
-                ops.recordTotalTime(req.method, status, now - start, classifierF(req)))))
-      }
-      resp <- client.run(req)
-      _ <- Resource.liftF(statusRef.set(Some(resp.status)))
-      end <- Resource.liftF(clock.monotonic(TimeUnit.NANOSECONDS))
-      _ <- Resource.liftF(ops.recordHeadersTime(req.method, end - start, classifierF(req)))
-    } yield resp)
-      .handleErrorWith { (e: Throwable) =>
+      resp <- (for {
```

Comment:
maybe put this in another method to improve readability

Reply:
@hamnis I broke the nesting out into functions. What do you think?